applications/RANSApplication/python_scripts/formulations/turbulence_models/scalar_turbulence_model_rans_formulation.py

```diff
@@ -0,0 +1,208 @@
+        CalculateNormalsOnConditions(self.GetBaseModelPart())
+
+        settings = self.GetParameters()
+
+        if (self.IsPeriodic()):
+            InitializePeriodicConditions(
+                self.GetBaseModelPart(),
+                self.GetModelPart(),
+                [self.GetSolvingVariable()])
+
+        linear_solver = GetKratosObjectType("LinearSolverFactory")(
+            settings["linear_solver_settings"])
+
+        builder_and_solver = CreateBlockBuilderAndSolver(
+            linear_solver,
+            self.IsPeriodic(),
+            self.GetCommunicator())
+
+        convergence_criteria = GetKratosObjectType("MixedGenericCriteria")([
+            (self.GetSolvingVariable(),
+             settings["relative_tolerance"].GetDouble(),
+             settings["absolute_tolerance"].GetDouble())])
+
+        if (self.is_steady_simulation):
+            scheme = self.scheme_type(settings["relaxation_factor"].GetDouble())
+        else:
+            scheme = GetKratosObjectType("BossakRelaxationScalarScheme")(
+                self.GetModelPart().ProcessInfo[Kratos.BOSSAK_ALPHA],
+                settings["relaxation_factor"].GetDouble(),
+                self.GetSolvingVariable())
+
+        self.solver = GetKratosObjectType("ResidualBasedNewtonRaphsonStrategy")(
+            self.GetModelPart(),
+            scheme,
+            convergence_criteria,
+            builder_and_solver,
+            settings["max_iterations"].GetInt(),
+            False,
+            False,
+            False)
+
+        self.solver.SetEchoLevel(self.echo_level)
+        convergence_criteria.SetEchoLevel(self.echo_level)
+
+        super().Initialize()
+        Kratos.Logger.PrintInfo(self.__class__.__name__, "Initialized formulation")
+
+    def SolveCouplingStep(self):
+        if (self.IsBufferInitialized()):
+            self.ExecuteBeforeCouplingSolveStep()
+            self.solver.Predict()
+            self.solver.SolveSolutionStep()
+            self.ExecuteAfterCouplingSolveStep()
+            Kratos.Logger.PrintInfo(self.__class__.__name__, "Solved  formulation.")
+            return True
+
+        return False
+
+    def GetStrategy(self):
+        return self.solver
+
+    def SetTimeSchemeSettings(self, settings):
+        if (settings.Has("scheme_type")):
+            scheme_type = settings["scheme_type"].GetString()
+            if (scheme_type == "steady"):
+                self.is_steady_simulation = True
+            elif (scheme_type == "bdf2" or scheme_type == "bossak"):
+                self.is_steady_simulation = False
+            else:
+                raise Exception(
+                    "Only \"steady\", \"bdf2\" and \"bossak\" scheme types supported. [ scheme_type = \""
+                    + scheme_type + "\" ]")
+        else:
+            raise Exception(
+                "\"scheme_type\" is missing in time scheme settings")
+
+    def GetMaxCouplingIterations(self):
+        return "N/A"
```

Comment:
I'd return 0. IMO a string variable is not the expected return type for `GetMaxCouplingIteration` method.

Reply:
done :)